src/evo/deterministicmns.cpp

```diff
@@ -223,10 +223,27 @@ std::vector<CDeterministicMNCPtr> CDeterministicMNList::GetProjectedMNPayees(int
     std::vector<CDeterministicMNCPtr> result;
     result.reserve(nCount);
 
+    auto remaining_hpmn_payments = 0;
     ForEachMNShared(true, [&](const CDeterministicMNCPtr& dmn) {
-        result.emplace_back(dmn);
+        if (dmn->pdmnState->nLastPaidHeight == nHeight) {
+            // We found the last MN Payee.
+            // If the last payee is a HPMN, we need to check its consecutive payments and pay him again if needed
+            if (dmn->nType == MnType::HighPerformance && dmn->pdmnState->nConsecutivePayments < dmn_types::HighPerformance.voting_weight) {
+                remaining_hpmn_payments = dmn_types::HighPerformance.voting_weight - dmn->pdmnState->nConsecutivePayments;
+                for ([[maybe_unused]] auto _ : irange::range(remaining_hpmn_payments)) {
+                    result.emplace_back(dmn);
+                }
+            }
+        }
+        return;
+    });
+
+    ForEachMNShared(true, [&](const CDeterministicMNCPtr& dmn) {
```

Comment:
don't we need to skip the current hpmn payee we found (and added) in the first loop (if any)? cause otherwise it feels like it's going to be added twice

Reply:
Well, no. Imagine a scenario with only 2 HPMNs in the network and a projection window of 10 blocks ahead?

Reply:
hm you got a point here. All right I'll come back with a change